dotCMS/src/main/java/com/dotcms/rest/api/v1/workflow/WorkflowResource.java

```diff
@@ -622,7 +623,66 @@ public final Response deleteAction(@Context final HttpServletRequest request,
 
         return response;
     } // deleteAction
+    
+    /**
+     * Change the order of the steps in a scheme
+     * @param request                           HttpServletRequest
+     * @param workflowReorderActionStepForm     WorkflowReorderBean
+     * @return Response
+     */
+    @PUT
+    @Path("/reorder/step/{stepId}/order/{order}")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response reorderStep(@Context final HttpServletRequest request,
+                                        @PathParam("stepId")   final String stepId, 
+                                        @PathParam("order")    final int order) {
+
+        final InitDataObject initDataObject = this.webResource.init
+                (null, true, request, true, null);
+        Response response;
 
+        try {
+            WorkflowStep step = workflowAPI.findStep(stepId);
+            WorkflowScheme scheme = workflowAPI.findScheme(step.getSchemeId());
+            List<WorkflowStep> steps = workflowAPI.findSteps(scheme);
+            IntStream.range(0, steps.size())
+                .filter(i -> steps.get(i).getId().equals(step.getId()))
+                .boxed()
+                .findFirst()
+                .map(i -> steps.remove((int) i));
+
+            int newOrder = (order > steps.size()) ? steps.size():order;
+            steps.add(newOrder, step);
+            
+            int i=0;
+            for(WorkflowStep stepp : steps) {
+                stepp.setMyOrder(i++);
+                workflowAPI.saveStep(stepp);
+            }
+            
+            response  = Response.ok(new ResponseEntityView("Ok")).build(); // 200
+        } catch (DoesNotExistException e) {
+
+            Logger.error(this.getClass(),
+                    "DoesNotExistException on reorderStep, stepId: " + stepId +
+                            ", exception message: " + e.getMessage(), e);
+            response = ExceptionMapperUtil.createResponse(e, Response.Status.NOT_FOUND);
+        } catch (Exception e) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)